### DISCUSSION TITLE
Print text in specific style to Word

### DISCUSSION THREAD

**Discussion by Rhys-Bergeron** [quarto-dev/quarto-cli#2657] [Q&A] [answered]
Is there a way to call out what style you want all the text to be printed to in Word. I want to create custom styles, but it seems the only ones that are available are the Headers.

**Comment by rgaiacs** [comment]
Hi @Rhys-Bergeron, Thanks for your question.

To customise the style of Word documents, you need to provide a reference document. See details at https://quarto.org/docs/output-formats/ms-word-templates.html.

> I want to create custom styles, but it seems the only ones that are available are the Headers.

Can you be more clear of what you are trying to do? I was able to change the font family. I had to change it in each style type as documented in https://quarto.org/docs/output-formats/ms-word-templates.html. It's a little time consuming but it does the job.

**Reply by Rhys-Bergeron** [reply]
Hey, thank you for the response. So let me be a little more specific. 

I'm aware of the feature of changing the Styles within the word document. I understand that you can change the Header 1 style and then when you would utilize 

`# Header1 
`

and it will be formatted to the specific style that you changed it to (Font Style, Color, Size, Placement, etc.)

My question is if I create a new style completely, lets say labeled "style1". Is there a way that I can reference that style when rendering to word. It appears there are only limited styles that are recognized through quarto. Namely, all the headers, the title, date, author, but I can't seem to find a way to access the custom styles that I have made in my Word template. 

Any thoughts?

**Reply by rgaiacs** [reply]
Quarto uses Pandoc when it converts the Markdown file to Word and the support to reference document is from Pandoc. Markdown has a limited set of elements. For example, syntax for the section header is

```
# Awesome
```

and the syntax for pre-formatted text is

```
`foo`
```

Pandoc's reference document has a style for most of the element defined in Markdown.

> My question is if I create a new style completely, lets say labeled "style1". Is there a way that I can reference that style when rendering to word. It appears there are only limited styles that are recognized through quarto.

From my knowledge, there is no way to tell Pandoc to apply a given style to an element when you are exporting to Word.

I don't know what are you trying to do but you could use as a workaround to use heading 6 to your needs. For example

```
###### Reserved for my custom style.
```

**Comment by cderv** [comment, marked as answer]
>  I want to create custom styles, but it seems the only ones that are available are the Headers.

First, each component in Markdown will be translated in a docx component with a default style apply to them. Best way to see what component (code, list, header, ...) is associated to what style is to add the component in a document, and then compile to docx and look in the resulting document which style is applied. You can use this document as based for tweaking and provide as non default reference doc.

If I understand correctly your use case is also creating new styles from scratch in your template docx document, and then you want to be able to use them (apply them) from your source `.qmd` file. 

If that is the case, you can do that for DOCX document using Custom style feature from Pandoc: https://pandoc.org/MANUAL.html#custom-styles. It appears we don't have mention to that in our doc and we should probably add it. 

Syntax would be for inline style 
````
[Get out]{custom-style="Emphatically"}, he said.
````
and 
````markdown
::: {custom-style="Poetry"}
| A Bird came down the Walk---
| He did not know I saw---
:::
````
for paragraph style

You can also reference existing style and not just new one. 

Hope it helps

**Reply by Rhys-Bergeron** [reply]
Perfect! Yes, this is exactly what I was looking for. Adding that documentation would be very helpful, I tried the syntax 

```
::: {custom-style="Poetry"}
A Bird came down the Walk---
He did not know I saw---
:::
```
and was not getting any luck. Mentioning the `|` would be helpful!.

**Reply by cderv** [reply]
>  Mentioning the | would be helpful!.

This is not part of the custom styles, it is just a usual Markdown syntax for keeping linebreak. Mentioned in here https://quarto.org/docs/authoring/markdown-basics.html#other-blocks

**Reply by Rhys-Bergeron** [reply]
Understood. My text that I was trying to create starts with a "1. " and the custom style was then ignored. Does placing the` |` also ensure that the custom style is not overwritten?

For example 

```
::: {custom-style="Poetry"}
1. Happy Birthday
:::
```

outputs 
![image](https://user-images.githubusercontent.com/63369818/193062217-bfcb2707-f93c-4106-b83a-3c31b9aa01c4.png)

but then produces the style that is desired. 

```
::: {custom-style="Poetry"}
| 1. Happy Birthday
:::
```

outputs  
![image](https://user-images.githubusercontent.com/63369818/193062380-231d4af1-cd96-4383-af35-6b5d120a4c26.png)

**Reply by cderv** [reply]
Is Poetry a custom styles ? 

This is Word styling matter here. When using `1.` you get List styling in Workd, I don't know if you can override that by anything else easily. 

By doing adding `|` you get a special other element that get another style. You should inspect style in word to understand how it is and how you can tweak in Word
